src/api/lib/authenticator.rb

```diff
@@ -147,14 +217,14 @@ def check_extracted_user
     unless @http_user
       if @login.blank?
         return true if check_for_anonymous_user
-        raise AuthenticationRequiredError.new
+        raise AuthenticationRequiredError
       end
-      raise AuthenticationRequiredError.new "Unknown user '#{@login}' or invalid password"
+      raise AuthenticationRequiredError, "Unknown user '#{@login}' or invalid password"
     end
 
     if @http_user.state == 'unconfirmed'
-      raise UnconfirmedUserError.new "User is registered but not yet approved. " +
-                                         "Your account is a registered account, but it is not yet approved for the OBS by admin."
+      raise UnconfirmedUserError, "User is registered but not yet approved. Your account " +
```

Comment:
@bgeuken all the exceptions raised in `Authenticator` get rescued here in `ApplicationController` so it renders a friendly error message to the user instead of 500

https://github.com/bgeuken/open-build-service/blob/add_kerberos_support/src/api/app/controllers/application_controller.rb#L215-L223